buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/apksize/GenerateVariantPlugin.groovy

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.firebase.gradle.plugins.measurement.apksize
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class GenerateVariantPlugin implements Plugin<Project> {
+
+    @Override
+    void apply(Project project) {
+        def childProjects = project.rootProject.childProjects
+        def firebaseProjectsNames = childProjects.findAll { isFirebaseProject(it.value) }.keySet()
+        project.android {
+            firebaseProjectsNames.each {
+                productFlavors.create("$it") {
+                    dimension "apkSize"
+                }
+            }
+            firebaseProjectsNames.each {
+                sourceSets.getByName(it) { sourceSet ->
+                    sourceSet.java.srcDirs += ['src/empty/java']
+                }
+            }
+        }
+        firebaseProjectsNames.each {
+            project.dependencies.add("${it}Implementation", project.rootProject.project(":${it}"))
+        }
+    }
```

Comment:
I think it would be clearer to have one loop containing the blocks instead of placing multiple loops within the configuration blocks. What do you think?

Reply:
Done. At one of the many iterations I needed it to get it to work, and then I didn't check back to see if it was necessary.